lib/chef-dk/policyfile/policyfile_location_specification.rb

```diff
@@ -0,0 +1,74 @@
+require "chef-dk/policyfile_lock"
+require "chef-dk/policyfile/local_lock_fetcher"
+require "chef-dk/policyfile/chef_server_lock_fetcher"
+
+module ChefDK
+  module Policyfile
+    class PolicyfileLocationSpecification
+
+      attr_reader :name
+      attr_reader :source_options
+      attr_reader :storage_config
+      attr_reader :chef_config
+      attr_reader :ui
+
+      LOCATION_TYPES = [:path, :server]
+
+      def initialize(name, source_options, storage_config, chef_config = nil)
+        @name = name
+        @source_options = source_options
+        @storage_config = storage_config
+        @ui = nil
+        @chef_config = chef_config
+      end
+
+      def revision_id
+        fetcher.lock_data["revision_id"]
+      end
+
+      def fetcher
+        @fetcher ||= begin
+                       if source_options[:path]
+                         Policyfile::LocalLockFetcher.new(name, source_options, storage_config)
+                       elsif source_options[:server]
+                         Policyfile::ChefServerLockFetcher.new(name, source_options, chef_config)
+                       else
+                         raise "Invalid policyfile lock location type"
```

Comment:
This should have a custom exception class and also give the list of valid location types.